### PR TITLE
Yocto recipe for Kanto Container Management

### PR DESCRIPTION
**Pull request by hristobojilov**
[#6] Yocto recipe for Kanto Container Management

Signed-off-by: Hristo Bozhilov <Hristo.Bozhilov@bosch.io>

### CODE REVIEW

**Comment by e-grigorov** [comment]
@hristobojilov The Eclipse Kanto project is currently restructuring its license scheme - from Eclipse Public License 2.0 only to dual licensing under Eclipse Public License 2.0 **and** Apache License 2.0 (you can read more about it [here](https://projects.eclipse.org/projects/iot.kanto/reviews/restructuring-review)).

This contribution will be merged into the project with this new licensing scheme so its important for the project and the community to have your agreement on proceeding with it given these new circumstances :blush: 

Please, provide your feedback as a comment here that you, as an author of this contribution, understand and agree with the change explained above. Thank you!

**Comment by hristobojilov** [comment]
> @hristobojilov The Eclipse Kanto project is currently restructuring its license scheme - from Eclipse Public License 2.0 only to dual licensing under Eclipse Public License 2.0 **and** Apache License 2.0 (you can read more about it [here](https://projects.eclipse.org/projects/iot.kanto/reviews/restructuring-review)).
> 
> This contribution will be merged into the project with this new licensing scheme so its important for the project and the community to have your agreement on proceeding with it given these new circumstances 😊
> 
> Please, provide your feedback as a comment here that you, as an author of this contribution, understand and agree with the change explained above. Thank you!

I understand and agree with the license change explained above.

File: recipes-containers/container-management/container-management.inc [hunk 1]
```diff
@@ -0,0 +1,16 @@
+# Contains configuration variables for Eclipse Kanto containers management
+
+# The destination directory of the containers management config file
+CM_CFG_DD ?= "${sysconfdir}"
+# The destination directory of the  containers management binaries
+CM_BIN_DD ?= "${bindir}"
+# The destination directory were the systemd service file for container management will be placed
+CM_SYSUNIT_DD ?= "${systemd_unitdir}/system"
+# Additional systemd units that are required by container management
+CM_SRV_REQ ?= "${@bb.utils.contains('PACKAGECONFIG', 'cloudconn', ' suite-connector.service', '',d)}"
```
Comment (e-grigorov) [inline recipes-containers/container-management/container-management.inc:10]: cloudconn - what is it about?

File: recipes-containers/container-management/container-management_git.bb [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+# to your conf/local.conf file used for the build configuration of your image
+# or you can configure it in a custom .bbappend file for example to disable it via adding
+#             PACKAGECONFIG:remove = "cloudconn"
+# into a .bbappend file to the original recipe.
+PACKAGECONFIG ?= "cloudconn"
+# This package feature enables cloud connectivity of the container management service via suite connector.
+# It's enabled by default as container management is expected to work as an integrated component of the suite connector and not as a
+# standalone service.
+# Nevertheless, if this is required - it can be disabled on demand as shown above.
+PACKAGECONFIG[cloudconn] = "--with-cloudconn,--without-cloudconn,,kanto/suite-connector"
```
Comment (e-grigorov) [inline recipes-containers/container-management/container-management_git.bb:44]: ,, -> ,

File: recipes-containers/container-management/files/service.template [hunk 3]
```diff
@@ -0,0 +1,14 @@
+[Unit]
+Description=Eclipse Kanto - Container Management
+Documentation=https://eclipse.org/kanto/docs/
+After=network.target containerd.service @CM_SRV_AFTER@
+Requires=network.target containerd.service @CM_SRV_REQ@
```
Comment (e-grigorov) [inline recipes-containers/container-management/files/service.template:5]: There is no such dependency to the suite-connector for all other recipes. I am not sure that we need this one here. Even more, Azure connector can be used.

File: recipes-containers/container-management/container-management_git.bb [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+           file://service.template \
+           "
+
+SRCREV = "${AUTOREV}"
+
+PV = "0.1.0-git${SRCPV}"
+
+GO_IMPORT = "github.com/eclipse-kanto/container-management"
+GO_INSTALL = "${GO_IMPORT}/containerm/daemon \
+             ${GO_IMPORT}/containerm/cli \
```
Comment (e-grigorov) [inline recipes-containers/container-management/container-management_git.bb:17]: The formatting can be fixed, see SRC_URI.

**Review by e-grigorov — CHANGES_REQUESTED** [review]

File: recipes-containers/container-management/container-management.inc [hunk 5]
```diff
@@ -0,0 +1,12 @@
+# Contains configuration variables for Eclipse Kanto container management
+
+# The destination directory of the containers management config file
+CM_CFG_DD ?= "${sysconfdir}"
+# The destination directory of the  containers management binaries
```
Comment (konstantina-gramatova) [inline recipes-containers/container-management/container-management.inc:5]: Please, align using containe**r** management everywhere.

File: recipes-containers/container-management/container-management.inc [hunk 6]
```diff
@@ -0,0 +1,16 @@
...
+CM_BIN_DD ?= "${bindir}"
+# The destination directory were the systemd service file for container management will be placed
+CM_SYSUNIT_DD ?= "${systemd_unitdir}/system"
+# Additional systemd units that are required by container management
+CM_SRV_REQ ?= "${@bb.utils.contains('PACKAGECONFIG', 'cloudconn', ' suite-connector.service', '',d)}"
+# Additional systemd units that  containers management must be started after
+CM_SRV_AFTER ?= "${@bb.utils.contains('PACKAGECONFIG', 'cloudconn', ' suite-connector.service', '',d)}"
+# The destination directory where the container management logs will be placed
+CM_LOD_DD ?= "${localstatedir}/log"
+# Controls cloud connectivity for containers via suite connector
```
[the remark this answers is not in the document]
Comment (konstantina-gramatova) [reply recipes-containers/container-management/container-management.inc:11]: Enables the digital twin representation for container management

maybe? : )

File: recipes-containers/container-management/container-management_git.bb [hunk 7]
```diff
@@ -0,0 +1,98 @@
...
+# Currently the cloudconn feature is enabled by default. If you would like to disable this feature you can add
+#             PACKAGECONFIG:remove:pn-container-management = "cloudconn"
+# to your conf/local.conf file used for the build configuration of your image
+# or you can configure it in a custom .bbappend file for example to disable it via adding
+#             PACKAGECONFIG:remove = "cloudconn"
+# into a .bbappend file to the original recipe.
+PACKAGECONFIG ?= "cloudconn"
+# This package feature enables cloud connectivity of the container management service via suite connector.
+# It's enabled by default as container management is expected to work as an integrated component of the suite connector and not as a
+# standalone service.
```
Comment (konstantina-gramatova) [inline recipes-containers/container-management/container-management_git.bb:42]: Update accordingly based on the definition in the .in file - Enables the digital twin .....

as an integrated component of the suite connector  -> e.g. as an integrated Kanto component

**Review by konstantina-gramatova — CHANGES_REQUESTED** [review]

File: recipes-containers/container-management/container-management.inc [hunk 8]
```diff
@@ -0,0 +1,12 @@
+# Contains configuration variables for Eclipse Kanto container management
+
+# The destination directory of the container management config file
+CM_CFG_DD ?= "${sysconfdir}"
+# The destination directory of the  container management binaries
```
Comment (konstantina-gramatova) [inline recipes-containers/container-management/container-management.inc:5]: of the  container management -> fix the two spaces here

File: recipes-containers/container-management/container-management.inc [hunk 9]
```diff
@@ -0,0 +1,12 @@
...
+CM_BIN_DD ?= "${bindir}"
+# The destination directory where the systemd service file for container management will be placed
+CM_SYSUNIT_DD ?= "${systemd_unitdir}/system"
+# The destination directory where the container management logs will be placed
+CM_LOD_DD ?= "${localstatedir}/log"
+# Enables the digital twin representation for container management
```
Comment (konstantina-gramatova) [inline recipes-containers/container-management/container-management.inc:11]: to finalize this description -> 

Enables the container management digital twin representation

File: recipes-containers/container-management/container-management_git.bb [hunk 10]
```diff
@@ -0,0 +1,98 @@
...
+# Different features of this package can be enabled depending on the target usage of the package.
+inherit pkgconfig
+# Currently the cloudconn feature is enabled by default. If you would like to disable this feature you can add
+#             PACKAGECONFIG:remove:pn-container-management = "cloudconn"
+# to your conf/local.conf file used for the build configuration of your image
+# or you can configure it in a custom .bbappend file for example to disable it via adding
+#             PACKAGECONFIG:remove = "cloudconn"
+# into a .bbappend file to the original recipe.
+PACKAGECONFIG ?= "cloudconn"
+# This package feature enables the digital twin representation for container management.
```
Comment (konstantina-gramatova) [inline recipes-containers/container-management/container-management_git.bb:40]: Update accordingly to ->


This package feature enables the container management digital twin representation.

**Review by konstantina-gramatova — CHANGES_REQUESTED** [review]

**Review by konstantina-gramatova — APPROVED** [review]

**Review by konstantina-gramatova — APPROVED** [review]

**Review by e-grigorov — APPROVED** [review]

**Pull request merged**